### PR TITLE
Maxpool revise

### PR DESCRIPTION
**Pull request by pszmel**
Overview:
 1. Added single layer tests
 2. Added unit tests
 3. Updated spec
 4. Removed second constructor and add default values to rounding_type and auto_pad to the first one

ticket: CVS-37455

### CODE REVIEW

File: docs/ops/pooling/MaxPool_1.md [hunk 1]
```diff
@@ -52,15 +52,16 @@
     * *floor*
   * **Type**: string
   * **Default value**: *floor*
+  * **Required**: *no*
```
Comment (ilyachur) [inline docs/ops/pooling/MaxPool_1.md:55]: @lazarevevgeny am I right that we have a ticket for the MO to make required all operation attributes?
Should we have the **Required** section in the spec?

File: docs/ops/pooling/MaxPool_1.md [hunk 2]
```diff
@@ -52,15 +52,16 @@
...
 
   * **Description**: *auto_pad* how the padding is calculated. Possible values:
-    * None (not specified): use explicit padding values.
+    * explicit: use explicit padding values.
     * *same_upper (same_lower)* the input is padded to match the output size. In case of odd padding value an extra padding is added at the end (at the beginning).
     * *valid* - do not use padding.
   * **Type**: string
-  * **Default value**: None
+  * **Default value**: *explicit*
   * **Required**: *no*
```
Comment (ilyachur) [inline docs/ops/pooling/MaxPool_1.md:65]: @lazarevevgeny If MO always generates this attribute, does it mean that it is required?
Or in general required means that we can work without this attribute

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/max_pool.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+std::vector<MaxPoolSpecificParams> max_pool_only_test_cases = {
+        MaxPoolSpecificParams{ { 1, 3, 32, 32 }, { 1, 1 },
+                               { 0, 0 }, { 0, 0 }, { 2, 2 },
+                               ngraph::op::RoundingType::FLOOR, ngraph::op::PadType::SAME_LOWER},
+        MaxPoolSpecificParams{ { 1, 3, 32, 32 }, { 1, 1 },
+                               { 0, 0 }, { 0, 0 }, { 2, 2 },
+                               ngraph::op::RoundingType::FLOOR, ngraph::op::PadType::SAME_UPPER},
+        MaxPoolSpecificParams{ { 1, 3, 32, 32 }, { 1, 1 },
+                               { 0, 0 }, { 0, 0 }, { 2, 2 },
+                               ngraph::op::RoundingType::CEIL, ngraph::op::PadType::SAME_UPPER},
```
Comment (ilyachur) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/max_pool.cpp:23]: Should we add tests for other PadType (EXPLICIT, SAME_LOWER, ...)?

File: docs/ops/pooling/MaxPool_1.md (hunk 1 above)
Comment (lazarevevgeny) [reply docs/ops/pooling/MaxPool_1.md:55]: That fact that MO will always generate all attributes does not mean that we should specify it here as required.

File: docs/ops/pooling/MaxPool_1.md (hunk 2 above)
Comment (lazarevevgeny) [reply docs/ops/pooling/MaxPool_1.md:65]: Nope. See the comment above

File: docs/ops/pooling/MaxPool_1.md [hunk 4]
```diff
@@ -52,15 +52,16 @@
     * *floor*
   * **Type**: string
   * **Default value**: *floor*
+  * **Required**: *no*
 
 * *auto_pad*
 
   * **Description**: *auto_pad* how the padding is calculated. Possible values:
-    * None (not specified): use explicit padding values.
+    * explicit: use explicit padding values.
```
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_1.md:60]: wrap word 'explicit' with double quotes

**Review by lazarevevgeny — APPROVED** [review]
Approved with one comment about the spec

**Review by pelszkow — APPROVED** [review]

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/max_pool.cpp (hunk 3 above)
Comment (pszmel) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/max_pool.cpp:23]: You are right. I will add it

**Review by ilyachur — APPROVED** [review]

**Review by ggalieroc — APPROVED** [review]

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/max_pool.cpp [hunk 5]
```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2020 Intel Corporation
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/max_pool.cpp:1]: There is max_pooling in `pooling.cpp`. Please, do not duplicate. review it (pooling.cpp)

**Review by iefode — CHANGES_REQUESTED** [review]

File: ngraph/test/backend/max_pool.in.cpp [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    auto f = make_shared<Function>(maxPool, ParameterVector{A});
+
+    std::vector<float> a{1, 2, 3, 4, 5, 6, 7, 8, 9};
+    std::vector<float> result{1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_input<float>({a});
+    test_case.add_expected_output<float>(out_shape, result);
+    test_case.run();
+}
```
Comment (jdanieck) [inline ngraph/test/backend/max_pool.in.cpp:187]: Missing newline character.

File: ngraph/test/type_prop/max_pool.cpp [hunk 7]
```diff
@@ -99,3 +99,18 @@ TEST(type_prop, max_pool_auto_padding_spatial_dims_dynamic)
...
+    const Shape pads_begin{0, 0};
+    const Shape pads_end{0, 0};
+    const Shape kernel_shape{2, 2};
+
+    auto arg = make_shared<op::Parameter>(element::f32, arg_shape);
+    auto mp = make_shared<op::v1::MaxPool>(arg, strides, pads_begin, pads_end, kernel_shape);
+
+    ASSERT_EQ(mp->get_rounding_type(), op::RoundingType::FLOOR);
+    ASSERT_EQ(mp->get_auto_pad(), op::PadType::EXPLICIT);
+}
```
Comment (jdanieck) [inline ngraph/test/type_prop/max_pool.cpp:116]: Missing newline character.

**Review by jdanieck — APPROVED** [review]

**Review by ggalieroc — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**